### ISSUE TITLE
Support for Greek numbering

### ISSUE THREAD

**Issue by rousbound** [typst/typst#3882]
### Description

Add the option "g" and "G" to numbering with greek letters:
```typst
set enum(numbering:"g)")
+ first
+ second
+ third
(...)
```

Would render to
```
sym.alpha first
sym.beta second
sym.gamma third
(...)
```

The same for uppercase.

### Use Case

When you have high levels of identation, we "only" have 1, a, A, i, I and some more "paragraph" like characters like sym.pilcrow and sym.section. It would be cool to have greek letters as an option, specially on deep nested documents to avoid repetition of the numbering types mentioned above.

I could create a package to create a wrapper for numbering for such a thing, but I think that it could be desirable and trivial to have the numbering pattern on the core Typst itself.

Edit:
Having in mind that western thinking is highly rooted in greek culture, I think this feature can be desirable for other language speaking users aside from greek ones too.

**Comment by Enivex** [comment]
Surely a Greek letter should be used if this is added, not `g` and `G`?

Alpha and beta are out, since the capital letters are indistinguishable from their latin counterparts. Gamma could work.

**Comment by rousbound** [comment]
You mean to use the greek numbering pattern in some way like this:
```typst
#set enum(numbering:"Gamma.i.1")

#set enum(numbering:"gamma.i.1")
```

Instead of:

```typst
#set enum(numbering:"G.i.1")
#set enum(numbering:"g.i.1")
```

I see your point, but if we are going to be more verbose then I think maybe its better like this:

```typst
#set enum(numbering:"Greek.i.1")
#set enum(numbering:"greek.i.1")
````

**Comment by Enivex** [comment]
No, I mean the actual Unicode symbols γ and Γ.

**Comment by Leedehai** [comment]
I prefer γ and Γ for greek counting numbers because (1) it doesn't take the place of "g"/"G" and (2) it aligns with precedents.

**Comment by rousbound** [comment]
Oh, ok, then I think you mean:
```typst
#set enum(numbering:"Γ.i.1")
#set enum(numbering:"γ.i.1")
```
The thing that crosses my mind is:
I would need to copy an paste a unicode symbol from docs/google or memorise unicode code and use the Ctrl+Shift+u method, right?

I'm not very used to this so it feels a lit bit too much of extra work at first sight.

But I see the point of "(2) aligns with precendent".

"(1) doesn't take the place of 'g/G'" I'm not sure why that could be a problem in the future, can you think of an example?

**Comment by Enivex** [comment]
> We would need to copy and paste a unicode symbol from docs/google or memorise unicode code and use the Ctrl+Shift+u method everytime, right?

There are various ways of inputing them depending on your setup. Regardless, it would be ergonomic for Greek users of typst, which should be the main priority.

**Comment by Leedehai** [comment]
> "(1) doesn't take the place of 'g/G'" I'm not sure why that could be a problem in the future, can you think of an example?

an example would be `#set enum(numbering:"group 1.")` :)

**Comment by Leedehai** [comment]
Taking a step back, should Typst support greek numbering natively? Note `numbering` can accept a user-provided function to map numbers to whatever strings they'd like.

As noted above, if Typst support greek numbering, γ/Γ should be the representative counter, and it's not the first character of the greek series and still seems like a break from precedents (and users might thought the Typst's greek counting series begins with γ).

**Comment by PgBiel** [comment]
> The thing that crosses my mind is: We would need to copy and paste a unicode symbol from docs/google or memorise unicode code and use the Ctrl+Shift+u method everytime, right?

No, you can write something like `"a" + sym.alpha + sym.gamma + "c"`

**Comment by Enivex** [comment]
> Using the alpha characters instead definitely would be more consistent.

The issue is that a capital alpha looks visually identical to a capital a. They're different codepoints, but still.

**Comment by PgBiel** [comment]
> > Using the alpha characters instead definitely would be more consistent.
> 
> The issue is that a capital alpha looks visually identical to a capital a. They're different codepoints, but still.

True. That could be mitigated by using `sym` in some manner, but I agree that can be confusing.

Perhaps the next best alternative could be using beta instead. Or even omega.

Gamma feels a bit arbitrary though (at least from a greek speaker's PoV, probably).

**Comment by Enivex** [comment]
> > > Using the alpha characters instead definitely would be more consistent.
> > 
> > The issue is that a capital alpha looks visually identical to a capital a. They're different codepoints, but still.
> 
> True. That could be mitigated by using `sym` in some manner, but I agree that can be confusing.
> 
> Perhaps the next best alternative could be using beta instead. Or even omega.
> 
> Gamma feels a bit arbitrary though (at least from a greek speaker's PoV, probably).

Beta has the same problem, though to a lesser extent, since the capital letter is indistinguishable from a capital b.

Gamma is the third letter in the Greek alphabet, and the first where both the lower and upper case versions are distinct.

Omega would be fine too (it's the last letter, and both are distinct)

**Comment by LuxxxLucy** [comment]
Hi all I just upload a PR for this issue. 

As for one of the major point of discussion of which char to use (or the symbol alpha), in the current version, I am still using the lowercase alpha for now, as I think it is more consistent. This can be changed if it provides better ergonomics so we should discuss.

Meanwhile I am mainly working on the numbering converting mechanism.  I think I need some help to proof reading the results as I am not a Greek Reader and my quick study of the greek numbering system might be seriously flawed. (Please refer to the expalnation and tests provided in #4273)

**Comment by MDLC01** [comment]
I haven't read the whole thread so I may be wrong, but I believe this issue was closed by #4273.
